Build break on OSX Mavericks (Travis)

```
src/src_libczmq_la-zsock.lo
src/zsock.c:591:43: error: format specifies type 'unsigned char' but the argument has type 'int' [-Werror,-Wformat]
            zmsg_addstrf (msg, "%" PRIu8, va_arg (argptr, int));
                                ~~~       ^~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/include/stdarg.h:35:29: note: expanded from macro 'va_arg'
#define va_arg(ap, type)    __builtin_va_arg(ap, type)
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/zsock.c:594:44: error: format specifies type 'unsigned short' but the argument has type 'int' [-Werror,-Wformat]
            zmsg_addstrf (msg, "%" PRIu16, va_arg (argptr, int));
                                ~~~        ^~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/include/stdarg.h:35:29: note: expanded from macro 'va_arg'
#define va_arg(ap, type)    __builtin_va_arg(ap, type)
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

https://travis-ci.org/libbitcoin/libbitcoin-server/jobs/46509578#L1301

OK, I'll fix these now.

On Sat, Jan 10, 2015 at 12:25 PM, Eric Voskuil <notifications@github.com>
wrote:

> src/src_libczmq_la-zsock.lo
> src/zsock.c:591:43: error: format specifies type 'unsigned char' but the argument has type 'int' [-Werror,-Wformat]
>             zmsg_addstrf (msg, "%" PRIu8, va_arg (argptr, int));
>                                 ~~~       ^~~~~~~~~~~~~~~~~~~~
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/include/stdarg.h:35:29: note: expanded from macro 'va_arg'
> #define va_arg(ap, type)    __builtin_va_arg(ap, type)
>                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
> src/zsock.c:594:44: error: format specifies type 'unsigned short' but the argument has type 'int' [-Werror,-Wformat]
>             zmsg_addstrf (msg, "%" PRIu16, va_arg (argptr, int));
>                                 ~~~        ^~~~~~~~~~~~~~~~~~~~
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/6.0/include/stdarg.h:35:29: note: expanded from macro 'va_arg'
> #define va_arg(ap, type)    __builtin_va_arg(ap, type)
>                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
> 2 errors generated.
>
> https://travis-ci.org/libbitcoin/libbitcoin-server/jobs/46509578#L1301
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/czmq/issues/888>.
>

--- a/src/zsock.c
+++ b/src/zsock.c
@@ -588,10 +588,10 @@ zsock_vsend (void *self, const char *picture, va_list argptr)
             zmsg_addstrf (msg, "%d", va_arg (argptr, int));
         else
         if (*picture == '1')
-            zmsg_addstrf (msg, "%" PRIu8, va_arg (argptr, int));
+            zmsg_addstrf (msg, "%" PRIu8, (uint8_t) va_arg (argptr, int));
         else
         if (*picture == '2')
-            zmsg_addstrf (msg, "%" PRIu16, va_arg (argptr, int));
+            zmsg_addstrf (msg, "%" PRIu16, (uint16_t) va_arg (argptr, int));
         else
         if (*picture == '4')
             zmsg_addstrf (msg, "%" PRIu32, va_arg (argptr, uint32_t));

